assertsprocessor/metrics.go

```diff
@@ -0,0 +1,166 @@
+package assertsprocessor
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+)
+
+type metricHelper struct {
+	logger                *zap.Logger
+	config                *Config
+	prometheusRegistry    *prometheus.Registry
+	latencyHistogram      *prometheus.HistogramVec
+	attributeValueRegExps *map[string]regexp.Regexp
+}
+
+func (p *metricHelper) compileSpanFilterRegexps() error {
+	p.logger.Info("consumer.Start compiling regexps")
+	for attName, matchExpString := range *p.config.AttributeExps {
+		compile, err := regexp.Compile(matchExpString)
+		if err != nil {
+			return err
+		}
+		(*p.attributeValueRegExps)[attName] = *compile
+	}
+	p.logger.Debug("consumer.Start compiled regexps successfully")
+	return nil
+}
+
+// Returns true if a span matches the span selection criteria
+func (p *metricHelper) shouldCaptureMetrics(span ptrace.Span) bool {
+	if len(*p.attributeValueRegExps) > 0 {
+		spanAttributes := span.Attributes()
+		for attName, matchExp := range *p.attributeValueRegExps {
+			value, found := spanAttributes.Get(attName)
+			if !found {
+				return false
+			}
+			//p.logger.Info("Found Span Attribute",
+			//	zap.String(attName, value.AsString()))
+
+			valueMatches := matchExp.String() == value.AsString() || matchExp.MatchString(value.AsString())
```

Comment:
Why do we need to consider `matchExp.String() == value.AsString()` as a match?

Reply:
This was done under the assumption that the request URI extracted from trace could be a template or have path parameters whereas the request context on server side will always be a template. 

Although we are using a regexp to extract the URI from trace, it can only remove query parameters. Templating path parameters is easy to do in the instrumentation on the client side but a difficult problem outside the client.